src/common/reduction/matching/GapMatcher.cpp

```diff
@@ -63,3 +71,133 @@ std::string GapMatcher::config(const std::string &prepend) const {
+
+  if (a1_b1_match && a2_b2_match && !a1_b2_match && !a2_b1_match) {
+    new_event_1.merge(new_cluster_a_1);
+    new_event_1.merge(new_cluster_b_1);
+    new_event_2.merge(new_cluster_a_2);
+    new_event_2.merge(new_cluster_b_2);
+  } else if (a1_b2_match && a2_b1_match && !a1_b1_match && !a2_b2_match) {
+    new_event_1.merge(new_cluster_a_1);
+    new_event_1.merge(new_cluster_b_2);
+    new_event_2.merge(new_cluster_a_2);
+    new_event_2.merge(new_cluster_b_1);
+  } else {
+    XTRACE(CLUSTER, DEB,
+           "Unable to match clusters into two distinct events, discarding "
+           "readouts");
+    Stats.DiscardedSpanTooLarge++;
+
+    //\todo count discarded multievents
+    return;
+  }
+  Stats.SplitSpanTooLarge++;
+  stash_event(new_event_1);
+  stash_event(new_event_2);
+  evt.clear();
+}
+
+void GapMatcher::split_cluster(Cluster cluster, Cluster *new_cluster_1,
+                               Cluster *new_cluster_2) {
+  sort_by_increasing_coordinate(cluster.hits);
+  uint last_coord = 0;
+  bool filled_cluster_1 = false;
+
+  for (Hit hit : cluster.hits) {
+    // no hits added to any clusters yet, automatically added to cluster 1
+    if ((new_cluster_1->empty())) {
+      last_coord = hit.coordinate;
+      new_cluster_1->insert(hit);
+      XTRACE(CLUSTER, DEB, "Adding first hit to cluster 1");
+    }
+    // cluster 1 not marked as full yet, and gap is small enough to still be
+    // cluster 1
+    else if ((hit.coordinate - last_coord <= minimum_coord_gap_) &&
+             (!filled_cluster_1)) {
+      last_coord = hit.coordinate;
+      new_cluster_1->insert(hit);
+      XTRACE(CLUSTER, DEB, "Adding another hit to cluster 1");
+    }
+    // cluster 1 not marked as full yet, and large enough gap to start new
+    // cluster hit goes into cluster 2
+    else if ((hit.coordinate - last_coord > minimum_coord_gap_) &&
+             (!filled_cluster_1)) {
+      last_coord = hit.coordinate;
+      filled_cluster_1 = true;
+      new_cluster_2->insert(hit);
+      XTRACE(CLUSTER, DEB, "Adding first hit to cluster 2");
+    }
+    // cluster 1 marked as full, and gap small enough to keep adding to cluster
+    // 2
+    else if ((hit.coordinate - last_coord <= minimum_coord_gap_) &&
+             (filled_cluster_1)) {
+      last_coord = hit.coordinate;
+      new_cluster_2->insert(hit);
+      XTRACE(CLUSTER, DEB, "Adding another hit to cluster 2");
+    }
+    // cluster 1 full, and gap large enough to finish cluster 2 and start a 3rd
+    // 3 or more clusters not supported, clusters are cleared and method exited
+    else if ((hit.coordinate - last_coord > minimum_coord_gap_) &&
+             (filled_cluster_1)) {
+      XTRACE(CLUSTER, DEB,
+             "More than 2 distinct clusters in plane, clearing clusters and "
+             "returning");
+      new_cluster_1->clear();
+      new_cluster_2->clear();
+      return;
+    }
+  }
+}
+
+bool GapMatcher::clusters_match(Cluster cluster_a, Cluster cluster_b) {
```

Comment:
might be good with an explanation here, also I think that probably some other code
would be better, like

abs(a-sum - b-sum) < tolerence (if I understood this correctly)